src/components/Register.jsx

```diff
@@ -1,12 +1,47 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
 
 const Register = () => {
-  const [name, setName] = useState("");
-  const [userName, setUserName] = useState("");
-  const [password, setPassword] = useState("");
+  const nagivate = useNavigate();
+  const [savedProfiles, setSavedProfiles] = useState([]);
+  const [formInputs, setFormInputs] = useState({
+    id: new Date().getTime(),
+    name: "",
+    firstName: "",
+    password: "",
+  });
+
+  // let savedProfiles = [formInputs];
+  useEffect(() => {
+    if (localStorage.getItem("profileData")) {
+      setSavedProfiles(JSON.parse(localStorage.getItem("profileData")));
+    }
+  }, []);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+
+    setFormInputs((prev) => {
+      return { ...prev, [name]: value };
+    });
+  };
 
   const handleSubmit = (e) => {
     e.preventDefault();
+    const isProfileExists = savedProfiles?.some(
+      (profile) => profile.name === formInputs.name
+    );
+
+    isProfileExists &&
```

Comment:
we can return here, in that case you won't have to check conditions in line 42,44